### ISSUE TITLE
Integrate Spectral Library Search (Proteomics)

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#2267]
A peptide spectral library is a curated, annotated database of LC-MS/MS peptide spectra. One essential utility of a peptide spectral library is to serve as consensus templates supporting the identification of peptide/proteins based on the correlation between the templates with experimental spectra.

One potential application of peptide spectral libraries is the identification of new, currently unknown mass spectra. Here, the spectra from the library are compared to the new spectra and if a match is found, the unknown spectra can be assigned the identity of the known peptide in the library.

Spectral library search has gained increased attention because of the availability of ever growing amounts of publicly available spectra.

This issue is intended to collect potential projects worth to pursue.

**Comment by timosachsenberg** [comment]
**Title:** Third-party tools integration for spectral library search

**Description (2-5 sentences):**

**Expected outcomes:**

**Required/Preferred skills:**
basic C++ knowledge, basic understanding of mass spectrometry and concepts of spectral library search

**Possible Mentors:** ???, ???

**Difficulty:** easy

**Comment by hroest** [comment]
to my knowledge, the current standard for spectral library search is SpectraST and it is important to consider that library search has many more aspects than just matching two spectra, one of the most important aspects are proper decoy spectra generation. 

The other question is whether we should first go for a SpectraST adapter before we try to implement stuff ourselves. Also, since SpectraST is in C++ we could look into linking against their library or using some of their code (if licences are compatible)

**Comment by timosachsenberg** [comment]
I agree that integration of the "gold standard" is a good thing. On the other hand, I had some discussions recently with people working on spectral library searches. They basically all agreed that the original SpectraST scoring is very outdated and can be improved upon.

**Comment by timosachsenberg** [comment]
We still have a prototype implementation of a SpectraSTAdapter in https://sourceforge.net/p/open-ms/code/HEAD/tree/branches/seybold/source/APPLICATIONS/TOPP/SpectraSTAdapter.C
I think it could be quickly brought to work with the current version.

**Comment by hroest** [comment]
in either case, it would be good to have the adaptor to compare results.

**Comment by Leon-Klee** [comment]
Is anyone still working on this? Our Immunology group is highly interested in this.

**Comment by lukaszimmermann** [comment]
We have SpectraSTSearchAdapter as an util. Would be nice to have people testing it.
